src/fish_indent.cpp

```diff
@@ -391,7 +404,29 @@ int main(int argc, char *argv[]) {
         }
     }
 
-    const wcstring src = read_file(stdin);
+    argc -= optind;
+    argv += optind;
+
+    wcstring src;
+    if (argc == 0) {
+        src = read_file(stdin);
+    } else if (argc == 1) {
+        FILE *fh = fopen(*argv, "r");
+        if (fh) {
+            src = read_file(fh);
+            fclose(fh);
+            if (output_location) {
+                output_location = *argv;
+            }
+        } else {
+            fwprintf(stderr, _(L"File could not be opened\n"), *argv);
```

Comment:
You printf string doesn't do anything with the argument you passing to it. Running `make lint` will help you find such mistakes.